core/src/OpenACC/Kokkos_OpenACC_ParallelReduce_MDRange.hpp

```diff
@@ -84,7 +89,14 @@ class Kokkos::Impl::ParallelReduce<CombinedFunctorReducerType,
         m_policy);
 
     reducer.final(&val);
-    *m_result_ptr = val;
+    if (m_result_ptr_on_device == false) {
+      acc_wait(async_arg);
```

Comment:
`acc_wait(async_arg)` is needed because the OpenACC reduction kernel can be executed asynchronously.
The OpenACC implementation supports only built-in Reducer types and thus `reducer.final()` is a no-op.

Reply:
Plz add a comment to that effect

Reply:
Added a comment